packages/ec2-metadata-service/src/MetadataService.ts

```diff
@@ -32,8 +32,10 @@ export class MetadataService {
   }
 
   async request(path: string, options: { method?: string; headers?: Record<string, string> }): Promise<string> {
-    const { endpoint, ec2MetadataV1Disabled } = await this.config;
-    const handler = new NodeHttpHandler();
+    const { endpoint, ec2MetadataV1Disabled, httpOptions } = await this.config;
+    const handler = new NodeHttpHandler({
+      requestTimeout: httpOptions?.timeout,
```

Comment:
should this timeout also be passed to connectionTimeout?

Reply:
functionally, the timeout error should be thrown with just `requestTimeout`, but we can add `connectionTimeout` as well that'll pick up the same value from options. it's better to catch the two different timeouts.